src/services/decoders/utils.ts

```diff
@@ -0,0 +1,60 @@
+import {DecoderOptions} from "../../typings/decoders";
+import {
+    ParsedKey,
+    REPLACEMENT_CHARACTER,
+} from "../../typings/formatters";
+import {
+    parseKey,
+    replaceDoubleBacklash,
+} from "../formatters/YscopeFormatter/utils";
+
+
+/**
+ * Preprocesses filter key by removing escaped backlash to facilitate simpler parsing, then parses
+ * the key.
+ *
+ * @param filterKey
+ * @return The parsed key.
+ */
+const preprocessThenParseFilterKey = (filterKey: string): ParsedKey => {
```

Comment:
(no need to address this right now - we can revisit in a future PR. we can file an issue if you also agree

With my recent experiences, i believe the Unicode Replacement character is used for replacing sequences that can't be decoded as UTF-8 characters, which means it may have a higher chance of occurrence than any other Unicode characters, depending on how users generate IRv2 logs.

In that case, we may want use a different character for replacement.

anyways, let's make sure this behaviour is covered when we put up the docs

Reply:
> the error will only trigger on the keys not the values, which are probably human generated, so less likely to use replacement character.

fair. a rare example of how this would happen is that people can write `SHIFT-JIS` Japanese in the keys and the logging framework / text editor decides to encode the log events as only UTF-8 in the logs. Then to select the key in the log viewer as a level / timestamp key, they would put the key partially (or fully) containing the unicode replacement character. again, i agree this is a multi-point failure, should be really rare, and for the existing use cases we're serving, likely we won't encounter such issues.

> we want to get rid of this eventually with antler parser

yeah, hopefully we won't really see odd use cases before we migrate there